api/src/main/java/org/openmrs/module/eptsreports/reporting/cohort/evaluator/CustomConfigurationsDataEvaluator.java

```diff
@@ -24,15 +24,30 @@ public DataSet evaluate(DataSetDefinition dataSetDefinition, EvaluationContext c
 
     CustomConfigurationsDataDefinition dsd = (CustomConfigurationsDataDefinition) dataSetDefinition;
 
-    Set<ProgramWorkflowState> programWorkflowStateSet = dsd.getWorkflowStates();
-
     SimpleDataSet dataSet = new SimpleDataSet(dataSetDefinition, context);
 
     DataSetRow row = new DataSetRow();
     row.addColumnValue(
         new DataSetColumn("report_time", "Report time", String.class),
         EptsReportUtils.formatDateWithTime(new Date()));
+    row.addColumnValue(
+        new DataSetColumn("states", "States", String.class), getWorkflowStates(context));
     dataSet.addRow(row);
     return dataSet;
   }
+
+  private String getWorkflowStates(EvaluationContext context) {
+    List<ProgramWorkflowState> workflowStateList =
+        (List<ProgramWorkflowState>) context.getParameterValue("state");
+    StringBuilder statesAsString = new StringBuilder();
+    String value = "";
+    for (ProgramWorkflowState workflowState : workflowStateList) {
+      value = workflowState.getConcept().getDisplayString();
+      statesAsString =
+          statesAsString.length() > 0
+              ? statesAsString.append(",").append(value)
+              : statesAsString.append(value);
+    }
```

Comment:
i had just finished testing my change when i saw this commit, seems slack has been off for some time else i had sent through some chats. i thought requesting me to work on this meant you were not going to deal with it @ningosi, am sorry we have had our time on the same thing, am actually still thinking out a better way around this than adding it as a column

Reply:
Seems some miscommunication here, I believe the idea was to work together on this issue to find a solution. Generally during report validation we have multiple people working on same report at same time, so best to clarify if you think this will be different

Reply:
yeah, somehow we worked on the same thing, i had the same approach which we had both agreed upon and just
when about to commit saw this commit and had to skip my commit, in a mean
time i invested in finding out a better approach towards this change.
adding a new `org.openmrs.module.reporting.common.ObjectUtil.format `for
`ProgramWorkflowState` seems to be the most straight forward approach.